### ISSUE TITLE
Generate build for multiple python versions for noarch python package

### ISSUE THREAD

**Issue by jsanjay63** [conda/conda-build#4029]
### Actual Behavior

Conda build output generated doesn't work with multiple python versions. I am trying to build conda noarch packages for multiple python versions like 3.6, 3.7, 3.8. The generated package is specific to only 1 version for eg: packagename-1.0-py38_0.tar or with build: noarch: python is like eg: packagename-1.0-py_0.tar. I tried different things like passing python argument as --python 3.6 and also tried with conda_build_config.yaml file. But, the package is generated only for py38. If I try installing this package with different python version, I get error on windows as: 
`-feature:/win-64::__cuda==10.1=0
Your installed CUDA driver is: 10.1`

### Expected Behavior

Conda build output artifact should work with multiple python versions for noarch.

##### `Output of conda info`

recipe meta.yaml snippet:
```
build:
  noarch: python/generic
  number: 0
  skip: True # [win]

requirements:
  host:
    - python
    - setuptools
  run:
    - python
    - aiohttp=3.6.2
    - contextily=1.0.0
    - deprecated=1.2.10
    - descartes=1.1.0
    - fastparquet=0.4.1
```

```
conda-build --python 3.6 --no-include-recipe recipe
```

**Comment by shoemakerdr** [comment]
The conda maintainers may need to correct me on this one... 

A thing to note on `noarch: python`: it is not tied to any python version. So if you build a noarch python package, it should work with any python versions you declare in your `meta.yaml`. However, this really should only be used for "pure" Python packages.

Another thing I noticed (which is mentioned in the [conda-build docs](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#architecture-independent-packages)), is using preprocessor-selectors with your noarch build:

```
build:
  noarch: python
  number: 0
  skip: True # [win].    <-- you shouldn't use preprocessor-selectors with noarch builds
```

If you have any compiled Python or Python that is architecture-dependent, you'll need to utilize [variants](https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#). It would probably look something like this:

```
# conda_build_config.yaml
python:
  - 3.6
  - 3.7
  - 3.8
```

```
# meta.yaml
build:
  number: 0

requirements:
  build:
    - python {{ python }}
    - setuptools
  host:
    - python
  run:
    - python
    - aiohttp=3.6.2
    - contextily=1.0.0
    - deprecated=1.2.10
    - descartes=1.1.0
    - fastparquet=0.4.1
```

Also, if you're planning on distributing this package as a library for other users, you may want to reconsider how strictly you're pinning the versions for your dependencies. This may cause some issues with other packages your downstream dependents might depend on.

**Comment by jsanjay63** [comment]
Thanks for your help. I was able to solve this issue with noarch: python. The issue was related the conda-build version, earlier I was referring to anaconda channel which had the latest 3.18.11. But, when I switched to conda-forge I build with the latest version 3.19.2. With 3.18.11, it was adding python_abi dependency for the specific python version which it was trying to build against. With 3.19.2, there is no more python_abi added dependency and this fixed my issue. Thanks
